tests/testthat/test-filter_keys.R

```diff
@@ -35,3 +35,29 @@ test_that("filter_keys() works", {
     connection_clean_up(conn)
   }
 })
+
+
+test_that("filter_keys() works with copy = TRUE", {
+  for (conn in get_test_conns()) {
+
+    x <- get_table(conn, "__mtcars")
+
+    filter <- x |>
+      utils::head(10) |>
+      dplyr::select(name) |>
+      dplyr::collect()
```

Comment:
suggestion
```
dplyr::pull(name)
```
and remove the 'pull' in expect_equal

Reply:
Note that `filter` is used twice in the `expect_equal` call, once in the `pull()` call and once in the `filter_keys()` call.

And it needs to be a data.frame in `filter_keys()`

Reply:
Also, you comment made me realize that we could improve the visible binding of the `filter_keys` tests, so I have done that now.